File: 7. Reverse Integer/ReverseInteger.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        abs_x = abs(x)
        result = 0
        while (abs_x  != 0):
            r = abs_x  % 10
            result = result * 10 + r
            abs_x  = abs_x  // 10
        if (x > 0 and result < 2 ** 31):
            return result
        elif (x < 0 and result <= 2 ** 31):
            return -result
        else:
            return 0
        

File: 7. Reverse Integer/ReverseInteger.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
	var result = 0, r = 0;
    while (x != 0) {
        r = x % 10;
        if (x > 0) {
            x = Math.floor(x / 10);
        } else {
            x = Math.ceil(x / 10);
        }
        result = result * 10 + r;
    }
    if (result >= 2147483647 || result <= -2147483648) return 0;
    return result;
    
};
